### PR TITLE
Replace deprecated API calls

### LINKED ISSUE #3: registryPackagesForQuery deprecated
We have been using your action for a while now to purge packages from GitHub. All of a sudden it stopped working.

It seems that the api endpoint that is used to search for packages is deprecated in the GiHub GraphQL API: https://developer.github.com/v4/breaking_changes/#changes-scheduled-for-2020-04-01 and they have been removed from the API completely, making this action no longer functional.

Instead we are instructed to use the `PackageOwner` object instead. For search, it is suggested to use `PackageSearch` - though I have not been able to find the latter yet.

It could be re-implemented using `repository.packages`, though it does not accept all the parameters that the original search does, so it's not a drop-in replacement unfortunately

### PR DESCRIPTION
**Pull request by reiniertimmer**
Closes  #3

### CODE REVIEW

**Comment by reiniertimmer** [comment]
@boredland are you interested in this?

At the moment your action is non-functional as far as I can tell

**Comment by boredland** [comment]
> @boredland are you interested in this?
> 
> At the moment your action is non-functional as far as I can tell

Of course! Will have a look later today!

**Comment by reiniertimmer** [comment]
Thanks 👍 

I am not 100% sure (not too familiar with the GraphQL API myself), but it seems like the original API calls got removed. Can you verify that?

**Comment by reiniertimmer** [comment]
Hi @boredland any updates on this? 😄

File: dist/index.js [hunk 1]
```diff
@@ -9772,8 +9774,13 @@ const run = async () => {
     let versionsToDelete = await getVersionsToDelete();
     let deletedVersions = [];
     for (const version of versionsToDelete) {
-        core.debug(`will try to delete ${JSON.stringify(version)}`);
-        await deleteVersion(version);
+        if (process.env.CI === 'true') {
```
Comment (boredland) [inline dist/index.js]: oh nice, we can test locally

File: index.js [hunk 2]
```diff
@@ -19,9 +19,13 @@ const getVersionsToDelete = async () => {
                         versions(last: ${versionLimit}){
                             edges {
                                 node {
-                                    id, 
-                                    updatedAt, 
+                                    id
                                     version
+                                    files(first: 1) {
```
Comment (boredland) [inline index.js:24]: what does the `first: 1` do? does that make sense?

**Review by boredland — APPROVED** [review]
thank you very much for helping out here! I had a deeper look into githubs gql docs (https://developer.github.com/v4/object/packageversion/) and this seems to be the current way to do things. Pretty bad from them to do such a breaking change on a important feature.

File: index.js (hunk 2 above)
Comment (reiniertimmer) [reply index.js:24]: Honestly, I'm not completely sure.

From files I got a list of file (all with the same date). I don't know what's in there really (my packages all had a single file, but I got a list of them anyway. Could be metadata). But they all had the same date, so I figured just to go with the first one

File: dist/index.js (hunk 1 above)
Comment (reiniertimmer) [reply dist/index.js]: That's the whole reason I put it in 😄 

> 
> 
> thank you very much for helping out here! I had a deeper look into githubs gql docs (https://developer.github.com/v4/object/packageversion/) and this seems to be the current way to do things. Pretty bad from them to do such a breaking change on a important feature.

Well, I agree.. partly

This was a preview API (deprecation was announced for April already). But if you're not on top of the API (and neither am I) these things are easy to miss.

Thanks for accepting this PR!

**Pull request merged**